client/webserver/site/src/js/settings.js

```diff
@@ -299,6 +302,47 @@ export default class SettingsPage extends BasePage {
     window.location.reload()
   }
 
+  /* Change application password  */
+  async changeAppPW () {
+    const page = this.page
+    Doc.hide(page.changePWErrMsg)
+
+    const clearValues = () => {
+      page.appPW.value = ''
+      page.newAppPW.value = ''
+      page.confirmNewPW.value = ''
+    }
+    // Ensure password fields are nonempty.
+    if (!page.appPW.value || !page.newAppPW.value || !page.confirmNewPW.value) {
+      page.changePWErrMsg.textContent = 'app password cannot be empty'
+      Doc.show(page.changePWErrMsg)
+      clearValues()
+      return
+    }
+    // Ensure password confirmation matches.
+    if (page.newAppPW.value !== page.confirmNewPW.value) {
+      page.changePWErrMsg.textContent = 'password confirmation doesn\'t match'
+      Doc.show(page.changePWErrMsg)
+      clearValues()
+      return
+    }
+    const loaded = app.loading(page.changeAppPW)
+    const req = {
+      appPW: page.appPW.value,
+      newAppPW: page.newAppPW.value
+    }
+    clearValues()
+    const res = await postJSON('/api/changeapppass', req)
+    if (!app.checkResponse(res, true)) {
```

Comment:
You can put `loaded()` in between these two lines so you'll only have to do it once.